packages/dds/map/src/map.ts

```diff
@@ -361,6 +361,12 @@ export class SharedMap extends SharedObject<ISharedMapEvents> implements IShared
         this.kernel.trySubmitMessage(content, localOpMetadata);
     }
 
+    protected rebaseOpCore(content: any, localOpMetadata: unknown) {
+        // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+        this.kernel.tryProcessMessage({ contents: content } as ISequencedDocumentMessage, false, localOpMetadata);
```

Comment:
I'm confused about calling tryProcessMessage here and casting to ISequencedDocumentMessage.
This is not real message, so we can't process it as if it were remote on. Why tryRebaseOp is not enough here?

Reply:
Without calling tryProcessMessage() here the local client would not see the changes from rebased ops since they are either resubmitted or processed as local ops.

Reply:
I'm confused. First, no matter what's the reason, we should not lie - this is not ISequencedDocumentMessage. I think at a minimum signature should change to reflect reality.

This method (rebaseOpCore) is for applying stashed local changed. tryRebaseOp should do the same on next level. 
There is no resubmission in this process (we are not connected yet), and no ack processing. So I'm still confused on why we call processing function. I think in each DDS we should only call tryRebaseOp that should do the right thing for this DDS (and same applies as we drill into layers).

For example, we may eventually get into applyMapClearMessage(). Is the problem that we need to call into submitMapClearMessage() instead (i.e. add one more step)?
Note that calling tryProcessMessage will land into process() ahndler and they assume that localOpMetadata !== undefined, but we can't have localOpMetadata - it's not serializable.

Let's chat about it, maybe I'm missing something obvious here.

Reply:
Wes walked me though and I'm wrong here. My suggestion - change argument type of tryProcessMessage to accept only content, and also change rebaseOp on messageHandlers to some other name, reflecting what it does, like generateLocalMetadataForStackedOp or something of that nature to more clearly articulate to engineers what functions do.